Assorted FreeBSD fixes

t6 opened the pull request:
## Description

* Allow modules that use evdev to build on FreeBSD.  This will allow building the GBM and Wayland platforms in addition to X11 on FreeBSD
* Some fixes for the Wayland platform on FreeBSD which didn't build anymore with some of the recent changes and waylandpp 0.2.2, because of missing headers for `wayland::presentation_t`, `wayland::xdg_wm_base_t`, and not linking with `libwayland-client-extra++`.
* FreeBSD 12.0 will switch to LLVM's lld on amd64 soon.  Unfortunately `find_soname` will break in that case due to slightly different linker output.  I'm not sure if this is appropriate for Linux so `cmake/scripts/linux/Macros.cmake` was copied for FreeBSD.
* Let Kodi build on ARM architectures on FreeBSD.  Tested on a Raspberry Pi 2 running FreeBSD 12.0-CURRENT a while ago.  Using the UI and a couple of addons worked fine, but when trying to play any audio/video file Kodi simply hangs right now.  But at least this allows one to build Kodi on FreeBSD/arm without jumping through any extra hoops.

I can break any of these out into separate pull requests if necessary.

<!--- Describe your change in detail -->

## Motivation and Context

I maintain the [multimedia/kodi-devel](https://www.freshports.org/multimedia/kodi-devel) port on FreeBSD.  The penultimate goal would be to build it without having to apply custom patches or at least reduce them to a minimum.

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?

As part of `multimedia/kodi-devel` updates on FreeBSD only.

<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Rechi commented on xbmc/platform/linux/input/LinuxInputDevices.cpp:
```diff
@@ -403,12 +410,16 @@ int CLinuxInputDevice::KeyboardGetSymbol(unsigned short value)
 */
   }
-
+#else
```
return for `TARGET_LINUX`  is now missing

Rechi commented on cmake/scripts/freebsd/ArchSetup.cmake:
```diff
@@ -1,4 +1,7 @@
 set(ARCH_DEFINES -D_LINUX -DTARGET_POSIX -DTARGET_FREEBSD)
+if(CORE_PLATFORM_NAME_LC STREQUAL rbpi)
+  list(APPEND ARCH_DEFINES -D_ARMEL -DTARGET_RASPBERRY_PI)
```
would be better fit into `cmake/platform/linux/rbpi.cmake`, like it is already done for aml and gbm

Rechi replied:
meant `cmake/platform/freebsd/rbpi.cmake`

t6 commented:
Hmm, the Jenkins build currently fails with this error which doesn't appear to be my fault.  Any chance to fix it?
```
ERROR: $GIT_COMMIT (d4540a9108a5740bd2799448333e3ee87f242ec8) doesn't match currently checked out revision (406db2b2961f355dc3e37112a8ff6de31efa9e82)
```

t6 commented:
I'm going to partially revert the Wayland fixes since they broke the Linux+Wayland build.  It seems there is some sort of problem with `generate-wayland-extra-protocols` on FreeBSD.

Rechi commented on cmake/platform/freebsd/gbm.cmake:
```diff
@@ -0,0 +1,4 @@
+set(PLATFORM_REQUIRED_DEPS OpenGLES EGL GBM LibDRM)
+set(PLATFORM_OPTIONAL_DEPS VAAPI)
+set(APP_RENDER_SYSTEM gles)
+list(APPEND PLATFORM_DEFINES -DMESA_EGL_NO_X11_HEADERS -DPLATFORM_SETTINGS_FILE=gbm.xml)
```
maybe consider to just include the gbm file from linux like it is done for wayland
https://github.com/xbmc/xbmc/blob/efe30507783ce1fd438df7d0a7d86459dc176585/cmake/platform/freebsd/wayland.cmake

Rechi commented on cmake/platform/freebsd/rbpi.cmake:
```diff
@@ -0,0 +1,3 @@
+set(PLATFORM_REQUIRED_DEPS OpenGLES EGL MMAL)
```
same here

lrusak commented:
For raspberry pi what is the point? AFAIK there I no hw accelerated video decode available on freebsd for the RPi.

yol reviewed (changes requested):
There is no need to link `wayland-client-extra++`.

Rechi commented on cmake/platform/freebsd/rbpi.cmake:
```diff
@@ -0,0 +1,2 @@
+include(cmake/platform/linux/rbpi.cmake)
+list(APPEND ARCH_DEFINES -D_ARMEL -DTARGET_RASPBERRY_PI)
```
`-D_ARMEL -DTARGET_RASPBERRY_PI` is now in cmake/platform/linux/rbpi.cmake so this can be removed

Commit:
Enable build of OSScreenSaverFreedesktop on FreeBSD

Signed-off-by: Tobias Kortkamp <tobik@FreeBSD.org>

Commit:
Create cmake/platform/linux/{gbm,rbpi}.cmake for FreeBSD

Signed-off-by: Tobias Kortkamp <tobik@FreeBSD.org>

Commit:
Allow Kodi to build on ARM architectures and also enable the rbpi platform on FreeBSD

Signed-off-by: Tobias Kortkamp <tobik@FreeBSD.org>

Commit:
Let WinSystemWayland build with Clang on FreeBSD

Signed-off-by: Tobias Kortkamp <tobik@FreeBSD.org>

Commit:
Fix find_soname when lld is the system linker

Signed-off-by: Tobias Kortkamp <tobik@FreeBSD.org>

t6 commented:
> There is no need to link wayland-client-extra++.

Well, Kodi will not link without it:
```
...
/usr/bin/ld: error: undefined symbol: wayland::xdg_toplevel_t::set_fullscreen(wayland::output_t)
>>> referenced by ShellSurfaceXdgShell.cpp
>>>               ShellSurfaceXdgShell.cpp.o:(KODI::WINDOWING::WAYLAND::CShellSurfaceXdgShell::SetFullScreen(wayland::output_t const&, float)) in archive build/windowing/wayland/windowing_WAYLAND.a

/usr/bin/ld: error: undefined symbol: wayland::xdg_toplevel_t::unset_fullscreen()
>>> referenced by ShellSurfaceXdgShell.cpp
>>>               ShellSurfaceXdgShell.cpp.o:(KODI::WINDOWING::WAYLAND::CShellSurfaceXdgShell::SetWindowed()) in archive build/windowing/wayland/windowing_WAYLAND.a
...
```
But this is probably related to the other Wayland trouble I mentioned earlier.  I've removed this from the PR too until I can figure out what the real problem is.  Clearly something else is going wrong. Maybe my `graphics/waylandpp` port is broken in some way.

Commit:
Fix build of Wayland platform on FreeBSD

cmake/scripts/freebsd/ExtraTargets.cmake is just a clone of Linux'
ExtraTargets.cmake, so just include it instead.  This way things
cannot go out of sync in the future.

t6 commented:
Right. This fixes it.

Commit:
Sync system/settings/freebsd.xml with linux.xml

DEBUG: CSettings: loaded settings definition from special://xbmc/system/settings/freebsd.xml
FATAL: Unable to load freebsd-specific settings definitions

Also fixes a crash when libva-intel-driver is installed

Signed-off-by: Tobias Kortkamp <tobik@FreeBSD.org>
